hack/add-version/main.go

```diff
@@ -203,120 +278,120 @@ func newVersionManager(ctx context.Context, region, bucket, distributionID strin
 	}, nil
 }
 
-func (m *versionManager) getMinorVersions(ctx context.Context, version string) (*update.VersionsList, error) {
-	baseVersion := semver.Major(version)
-	return m.getVersions(ctx, baseVersion)
-}
-
-func (m *versionManager) getPatchVersions(ctx context.Context, version string) (*update.VersionsList, error) {
-	baseVersion := semver.MajorMinor(version)
-	return m.getVersions(ctx, baseVersion)
-}
-
-func (m *versionManager) addMinorVersion(ctx context.Context, version string, minorVersions *update.VersionsList) error {
-	baseVersion := semver.Major(version)
-	minorVersion := semver.MajorMinor(version)
-	return m.addVersion(ctx, baseVersion, minorVersion, minorVersions)
-}
-
-func (m *versionManager) addPatchVersion(ctx context.Context, version string, patchVersions *update.VersionsList) error {
-	baseVersion := semver.MajorMinor(version)
-	patchVersion := semver.Canonical(version)
-	return m.addVersion(ctx, baseVersion, patchVersion, patchVersions)
-}
-
-func (m *versionManager) getVersions(ctx context.Context, baseVersion string) (*update.VersionsList, error) {
-	granularity, err := granularityFromVersion(baseVersion)
-	if err != nil {
-		return nil, err
-	}
-	out, err := m.s3c.GetObject(ctx, &s3.GetObjectInput{
+func (m *versionManager) getVersionList(ctx context.Context, ver version, gran granularity) (*versionsapi.List, error) {
+	in := &s3.GetObjectInput{
 		Bucket: aws.String(m.bucket),
-		Key:    aws.String(versionJSONPath(granularity, baseVersion)),
-	})
-	if err != nil {
-		var nosuchkey *s3types.NoSuchKey
-		if errors.As(err, &nosuchkey) {
-			return nil, errVersionListMissing
-		}
+		Key:    aws.String(ver.JSONPath(gran)),
+	}
+	out, err := m.s3c.GetObject(ctx, in)
+	var noSuchkey *s3types.NoSuchKey
+	if errors.As(err, &noSuchkey) {
+		return nil, errVersionListMissing
+	} else if err != nil {
 		return nil, err
 	}
 	defer out.Body.Close()
-	var versions update.VersionsList
-	if err := json.NewDecoder(out.Body).Decode(&versions); err != nil {
+
+	var list versionsapi.List
+	if err := json.NewDecoder(out.Body).Decode(&list); err != nil {
 		return nil, err
 	}
-	return &versions, nil
+
+	return &list, nil
 }
 
-func (m *versionManager) addVersion(ctx context.Context, baseVersion, version string, list *update.VersionsList) error {
-	granularity, err := granularityFromVersion(baseVersion)
-	if err != nil {
-		return err
-	}
-	list.Versions = append(list.Versions, version)
+func (m *versionManager) updateVersionList(ctx context.Context, list *versionsapi.List) error {
 	sort.Strings(list.Versions)
```

Comment:
Should be [semver sort](https://pkg.go.dev/golang.org/x/mod/semver#Sort). My bad :)

```suggestion
	semver.Sort(list.Versions)
```